source.go

```diff
@@ -712,8 +714,82 @@ func fillDefaults(s *Source) {
 		fillDefaults(&s.Build.Source)
 	case s.Inline != nil:
 	}
+
+}
+
+func (git *SourceGit) fillDefaults(generators []*SourceGenerator) {
+	if git == nil {
+		return
+	}
+
+	u, err := url.Parse(git.URL)
+	if err != nil {
+		// This should be validated elsewhere
+		return
+	}
+
+	for _, generator := range generators {
+		generator.fillDefaults(u.Host, &git.Auth)
+	}
 }
 
+func (g *SourceGenerator) fillDefaults(host string, authInfo *GitAuth) {
+	if g == nil || authInfo == nil {
+		return
+	}
+
+	switch {
+	case g.Gomod != nil:
+		g.Gomod.fillDefaults(host, authInfo)
+	}
+}
+
+func (gm *GeneratorGomod) fillDefaults(host string, authInfo *GitAuth) {
+	var (
+		initialize = func() {
+			if gm.Auth == nil {
+				gm.Auth = make(map[string]GomodGitAuth)
+			}
+		}
+
+		noop = func() {}
+
+		initOnce = func() {
```

Comment:
This seems a bit overzealous here.

Reply:
Fixed. Can just init right before assiging.

TIL you can read the value and bool from nil go maps (will always be the zero-value and false) but writing to them panics.